dropwizard-logging/src/test/java/io/dropwizard/logging/FileAppenderFactoryTest.java

```diff
@@ -125,7 +132,45 @@ public void hasMaxFileSizeValidation() throws Exception{
         errors = ConstraintViolations.format(validator.validate(fileAppenderFactory));
         assertThat(errors).isEmpty();
     }
+    
+    @Test
+    public void testCurrentFileNameErrorWhenArchiveIsNotEnabled() throws Exception {
+        FileAppenderFactory fileAppenderFactory = new FileAppenderFactory();
+        fileAppenderFactory.setArchive(false);
+        ImmutableList<String> errors =
+                ConstraintViolations.format(validator.validate(fileAppenderFactory));
+        assertThat(errors)
+                .containsOnly("currentLogFilename can only be null when archiving is enabled");
+        fileAppenderFactory.setCurrentLogFilename("test");
+        errors = ConstraintViolations.format(validator.validate(fileAppenderFactory));
+        assertThat(errors).isEmpty();
+    }
+    
+    @Test
+    public void testCurrentFileNameCanBeNullWhenArchiveIsEnabled() throws Exception {
+        FileAppenderFactory fileAppenderFactory = new FileAppenderFactory();
+        fileAppenderFactory.setArchive(true);
+        fileAppenderFactory.setArchivedLogFilenamePattern("name-to-be-used");
+        fileAppenderFactory.setCurrentLogFilename(null);
+        ImmutableList<String> errors =
+                ConstraintViolations.format(validator.validate(fileAppenderFactory));
+        assertThat(errors).isEmpty();
+    }
+
+    @Test
+    public void testCurrentLogFileNameIsEmptyAndAppenderUsesArchivedNameInstead() throws Exception {
```

Comment:
How do you feel about the following implementation instead (it's a bit shorter):

```java
@Test
public void testCurrentLogFileNameIsEmptyAndAppenderUsesArchivedNameInstead() throws Exception {
    final FileAppenderFactory<ILoggingEvent> appenderFactory = new FileAppenderFactory<>();
    appenderFactory.setArchivedLogFilenamePattern(folder.newFile("test-archived-name-%d.log").toString());
    final FileAppender<ILoggingEvent> rollingAppender = appenderFactory.buildAppender(new LoggerContext());

    final String file = rollingAppender.getFile();
    final String dateSuffix = LocalDateTime.now().format(DateTimeFormatter.ofPattern("YYYY-MM-dd"));
    final String name = Files.getNameWithoutExtension(file);
    Assert.assertEquals("test-archived-name-" + dateSuffix, name);
}
```